intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/verifiers/PluginIdVerifier.kt

```diff
@@ -0,0 +1,57 @@
+package com.jetbrains.plugin.structure.intellij.verifiers
+
+import com.jetbrains.plugin.structure.base.problems.PropertyNotSpecified
+import com.jetbrains.plugin.structure.intellij.beans.PluginBean
+import com.jetbrains.plugin.structure.intellij.plugin.CORE_PLUGIN_ID
+import com.jetbrains.plugin.structure.intellij.plugin.SPECIAL_IDEA_PLUGIN_ID
+import com.jetbrains.plugin.structure.intellij.problems.IllegalPluginIdPrefix
+import com.jetbrains.plugin.structure.intellij.problems.PropertyWithDefaultValue
+import com.jetbrains.plugin.structure.intellij.problems.TemplateWordInPluginId
+
+val DEFAULT_ILLEGAL_PREFIXES = listOf("com.example", "net.example", "org.example", "edu.example", "com.intellij", "org.jetbrains")
+
+val PRODUCT_ID_RESTRICTED_WORDS = listOf(
+  "clion",  "datagrip", "datalore", "dataspell", "dotcover", "dotmemory", "dotpeek", "dottrace", "fleet", "goland",
+  "intellij", "phpstorm", "pycharm", "resharper", "rider", "rubymine", "space", "webstorm", "youtrack",
+)
+
+class PluginIdVerifier {
+
+  fun verify(plugin: PluginBean, descriptorPath: String, problemRegistrar: ProblemRegistrar) {
+    val id = plugin.id ?: return
+
+    when {
+      id.isBlank() -> {
+        problemRegistrar.registerProblem(PropertyNotSpecified("id"))
+      }
+      "com.your.company.unique.plugin.id" == id -> {
+        problemRegistrar.registerProblem(PropertyWithDefaultValue(descriptorPath, PropertyWithDefaultValue.DefaultProperty.ID, id))
+      }
+      else -> {
+        validatePropertyLength("id", id, MAX_PROPERTY_LENGTH, descriptorPath, problemRegistrar)
+        validateNewlines("id", id, descriptorPath, problemRegistrar)
+        verifyPrefix(plugin, descriptorPath, problemRegistrar)
+      }
+    }
+  }
+
+  private fun verifyPrefix(plugin: PluginBean, descriptorPath: String, problemRegistrar: ProblemRegistrar) {
+    if (isDevelopedByJetBrains(plugin)) {
```

Comment:
Shouldn't it be `isDevelopedByJetBrains` from `PluginVendors`?

Reply:
This is in two parallel PRs that are in review concurrently (see #952). This will be consolidated into a single codebase in the subsequent cleanup.